### ISSUE TITLE
Python opencv net.forwardAsync doesn't return all output layers

### ISSUE THREAD

**Issue by jbrownkramer** [opencv/opencv#20669]
##### System information (version)

- OpenCV => 4.5
- Operating System / Platform => Windows
- Compiler => Pip installed

##### Detailed description

In python, net.forwardAsync is only returning one output layer from a yolo network.

##### Steps to reproduce

```
net = cv2.dnn.readNetFromDarknet(cfg, weights)
net.setPreferableBackend(cv2.dnn.DNN_BACKEND_INFERENCE_ENGINE)
net.setInput(blob)
a = net.forwardAsync()

while not a.valid():
  time.sleep(1/1000)

print(a.get().shape)
```

prints `(768, 16)`, but in fact the output should be two blobs, one of shape (192, 16) and one of shape (768, 16).

Furthermore, I cannot pass in an array of layer names, and, strangely, if I pass in the name of the smaller output layer, a.valid() never returns true.

**Comment by alalek** [comment]
> Pip installed

How is it installed?
AFAIK, `opencv-python` package doesn't really support `DNN_BACKEND_INFERENCE_ENGINE`.

What model is used?
What behavior is observed without `.setPreferableBackend()` call?

**Comment by jbrownkramer** [comment]
> > Pip installed
> 
> How is it installed?
> AFAIK, `opencv-python` package doesn't really support `DNN_BACKEND_INFERENCE_ENGINE`.

You are right.  I had a momentary mental lapse when I wrote that.  I will fix it in the problem description.  What I am using is the build of OpenCV that comes with the (Windows) install of OpenVINO

> 
> What model is used?

This is a tiny-yolo v4 with relu activations.  It's using the config and weights formats from https://github.com/AlexeyAB/darknet

> What behavior is observed without `.setPreferableBackend()` call?

I'll get back to you on that one.

**Comment by jbrownkramer** [comment]
> What behavior is observed without `.setPreferableBackend()` call?

With OpenCV built against OpenVINO, the default backend is InferenceEngine, so removing the `.setPreferableBackend()` call doesn't change anything.

With the OpenCV from `pip install opencv-python`, we get the error `Asynchronous forward is supported for Inference Engine backends only in function 'cv::dnn::dnn4_v20201117::Net::forwardAsync'`

**Comment by berak** [comment]
> In python, net.forwardAsync is only returning one output layer from a yolo network.

this seems to be true in c++, too. [AsyncArray](https://docs.opencv.org/master/d4/d20/classcv_1_1AsyncArray.html) can only contain a singe Mat.

makes me think, that [the c++ sample here](https://github.com/opencv/opencv/blob/c703f1eed6c3da4f14b37cc151e03463bb4ff54f/samples/dnn/object_detection.cpp#L236-L240) is flawed 
(it only takes the bottom layer of the yolo scale pyramid)

**Comment by Suaro** [comment]
> ##### System information (version)
> * OpenCV => 4.5
> * Operating System / Platform => Windows
> * Compiler => This is the version of OpenCV that comes with OpenVINO.  If it is compiling anything, it's probably using Visual Studio.
> 
> ##### Detailed description
> In python, net.forwardAsync is only returning one output layer from a yolo network.
> 
> ##### Steps to reproduce
> ```
> net = cv2.dnn.readNetFromDarknet(cfg, weights)
> net.setPreferableBackend(cv2.dnn.DNN_BACKEND_INFERENCE_ENGINE)
> net.setInput(blob)
> a = net.forwardAsync()
> 
> while not a.valid():
>   time.sleep(1/1000)
> 
> print(a.get().shape)
> ```
> 
> prints `(768, 16)`, but in fact the output should be two blobs, one of shape (192, 16) and one of shape (768, 16).
> 
> Furthermore, I cannot pass in an array of layer names, and, strangely, if I pass in the name of the smaller output layer, a.valid() never returns true.


Same issue with this [OpenVINO net ](https://docs.openvino.ai/2019_R1/_head_pose_estimation_adas_0001_description_head_pose_estimation_adas_0001.html) in C++. Its return three output layers "angle_y_fc", "angle_p_fc"  and "angle_r_fc". forwardAsync() only returns last layer. Any updates on this issue?
